### PR TITLE
set random seed compliant with tf2+

### PR DESCRIPTION
**Pull request by psteinb**
see also https://www.tensorflow.org/api_docs/python/tf/random/set_seed

### CODE REVIEW

**Comment by psteinb** [comment]
we might want to check if the random seed could be set through the keras API ... as this might be more stable as `tf.random`
